docs/spec/typeddict.rst

```diff
@@ -637,3 +637,193 @@ In particular allowing ``Annotated[]`` to be the outermost annotation
 for an item allows better interoperability with non-typing uses of
 annotations, which may always want ``Annotated[]`` as the outermost annotation
 (`discussion <https://bugs.python.org/issue46491>`__).
+
+
+Read-only Items
+===============
+
+(Originally specified in :pep:`705`.)
+
+``typing.ReadOnly`` type qualifier
+----------------------------------
+
+The ``typing.ReadOnly`` type qualifier is used to indicate that an item declared in a ``TypedDict`` definition may not be mutated (added, modified, or removed)::
+
+    from typing import ReadOnly
+
+    class Band(TypedDict):
+        name: str
+        members: ReadOnly[list[str]]
+
+    blur: Band = {"name": "blur", "members": []}
+    blur["name"] = "Blur"  # OK: "name" is not read-only
+    blur["members"] = ["Damon Albarn"]  # Type check error: "members" is read-only
+    blur["members"].append("Damon Albarn")  # OK: list is mutable
+
+
+Alternative functional syntax
+-----------------------------
+
+The :pep:`alternative functional syntax <589#alternative-syntax>` for TypedDict also supports the new type qualifier::
```

Comment:
Can you link this to https://typing.readthedocs.io/en/latest/spec/typeddict.html#alternative-syntax instead? It would be even better if that section directly mentioned `ReadOnly`.